tools/slicec-cs/src/generators/exception_generator.rs

```diff
@@ -44,118 +56,144 @@ pub fn generate_exception(exception_def: &Exception) -> CodeBlock {
             .into(),
     );
 
-    exception_class_builder.add_block(
-        format!("private static readonly string SliceTypeId = typeof({exception_name}).GetSliceTypeId()!;").into(),
+    builder.add_block(
+        format!("private static readonly string SliceTypeId = typeof({class_name}).GetSliceTypeId()!;").into(),
     );
 
-    if !exception_def.all_fields().is_empty() {
-        // Parameterless constructor.
-        exception_class_builder.add_block(
-            FunctionBuilder::new("public", "", &exception_name, FunctionType::BlockBody)
-                .add_comment(
-                    "summary",
-                    format!(r#"Constructs a new instance of <see cref="{}"/>."#, &exception_name),
-                )
-                .build(),
-        );
-
-        exception_class_builder.add_block(one_shot_constructor(exception_def));
-
-        // TODO: generate secondary constructor like for classes.
+    let constructor_summary = format!(r#"Constructs a new instance of <see cref="{class_name}" />."#);
+
+    if !all_fields.is_empty() {
+        // parameterless constructor.
+        // The constructor needs to be public for System.Activator.CreateInstance.
+        let mut parameterless_constructor = FunctionBuilder::new("public", "", &class_name, FunctionType::BlockBody);
+        parameterless_constructor.add_comment("summary", constructor_summary.clone());
```

Comment:
The clone should be unnecessary here, you can probably just pass a borrow:
```suggestion
        parameterless_constructor.add_comment("summary", &constructor_summary);
```
This should be updated in the class generator as well.